stl/inc/ranges

```diff
@@ -3068,6 +3068,19 @@ namespace ranges {
         requires view<_Vw> && input_range<range_reference_t<_Vw>>
     class join_view : public _Join_view_base<_Vw> {
         // clang-format on
+
+#ifndef _USE_JOIN_VIEW_INPUT_RANGE
+        static_assert(forward_range<_Vw> || borrowed_range<_Vw>,

```

Comment:
I don't think borrowedness matters here? The example showcasing the issue uses a `subrange<regex_iterator>`, which is very much borrowed.

Reply:
Gah, I meant `borrowed_range<range_reference_t<_Vw>>`.  If the iterators can outlive the range, surely they can outlive some iterator we dereferenced to get that range. Let's back off until I have time to think through this.
```suggestion
        static_assert(forward_range<_Vw>,
```

Reply:
I am beginning to believe it was a mistake to apply the _`ref-is-glvalue`_ optimization to input ranges at all.